hazelcast/src/main/java/com/hazelcast/nio/IOUtil.java

```diff
@@ -388,6 +391,89 @@ public static File getFileFromResources(String resourceFileName) {
         }
     }
 
+    /**
+     * Deep copies source to target and creates the target if necessary. Source can be a directory or a file and the target
+     * can be a directory or file. If the source is a directory, expects that the target is a directory (or that it doesn't exist)
+     * and nests the copied source under the target directory.
+     *
+     * @param source the source
+     * @param target the destination
+     * @throws IllegalArgumentException if the source was not found or the source is a directory and the target is a file
+     * @throws HazelcastException       if there was any exception while creating directories or copying
+     */
+    public static void copy(File source, File target) {
+        if (!source.exists()) {
+            throw new IllegalArgumentException("Source does not exist");
+        }
+        if (source.isDirectory()) {
+            copyDirectory(source, target);
+        } else {
+            copyFile(source, target, -1);
+        }
+    }
```

Comment:
I know it's a pain in the neck, but these would need some integration tests to make sure it works.